lib/bk/commands/artifacts.rb

```diff
@@ -103,26 +104,23 @@ def call(args: {}, url_or_slug: nil, **options)
           has_next_page = build.jobs.page_info.has_next_page
 
           jobs = build.jobs.edges.map(&:node)
-          Parallel.each(jobs) do |job|
+          bar = ProgressBar.create(total: jobs.count, throttle_rate: 1, format: '%a %t [%c/%C BK jobs]: %B %j%%, %E')
+
+          Parallel.each(jobs, in_threads: 8) do |job|
```

Comment:
This needs to be `in_threads` because otherwise the default is `in_processes`. When splitting each into a separate process, I believe each is getting a separate copy of the `bar` object. As a result, the `increment` function would cause a bar to print to the scene with progress *related to that one process*, but there wouldn't be any shared memory about what the current progress is. As a result, the output was nonsensically jumping around.

By making this in threads instead, each thread will share the same bar in memory, so the progress increases in an expected linearly sequential way. Since this process is bound by network (downloading from BK) and file IO (writing files to disk) and NOT CPU (nothing here is necessarily CPU intensive), moving to threads still keeps us moving fast (also spawns fewer ruby processes).